os/arch/arm/src/common/up_createstack.c

```diff
@@ -243,6 +260,14 @@ int up_create_stack(FAR struct tcb_s *tcb, size_t stack_size, uint8_t ttype)
 		heapinfo_exclude_stacksize(tcb->stack_alloc_ptr);
 #endif
 		board_led_on(LED_STACKCREATED);
+#ifdef CONFIG_MPU_STACKGUARD
+		/* The smallest size that can be programmed for an MPU region is 32 bytes */
+		tcb->stack_guard = (uint32_t)tcb->stack_alloc_ptr + MPU_REGION_SIZE_ALIGNMENT;
+		mpu_configure_app_regs(&tcb->mpu_regs[STACKGUARD_MPU_NUM * REG_MAX], STACKGUARD_MPU_NUM, tcb->stack_guard, MPU_REGION_SIZE_ALIGNMENT, true, false);
+		/* Adjust the adj_stack_size after guard_size calculationn */
+		tcb->adj_stack_size = tcb->adj_stack_size - MPU_REGION_SIZE_ALIGNMENT;
```

Comment:
Why do you subtract `MPU_REGION_SIZE_ALIGNMENT`?

Reply:
The exact stack size available for task will be reduced by 32 byte because of guard memory used for protection.